modules/test/playwright/tests/frontend-data-set-admin-web/tests/data-set-fragment/dataSets.spec.ts

```diff
@@ -25,97 +29,442 @@ export const test = mergeTests(
 	fdsFragmentPageTest
 );
 
-let dataSetERC: string;
-let dataSetLabel: string;
+const dataSetERCs = [];
+let article;
+let dataSetERC;
+let dataSetLabel;
+let siteId;
+let structuredContentId;
+let structuredContentTitle;
+
+const adminUserDataSetConfig = {
+	erc: getRandomString(),
+	label: getRandomString(),
+	restApplication: '/headless-admin-user/v1.0',
+	restEndpoint: '/v1.0/roles',
+	restSchema: 'Role',
+};
+
+const structuredContentDataSetConfig = {
+	erc: getRandomString(),
+	label: getRandomString(),
+	restApplication: '/headless-delivery/v1.0',
+	restEndpoint: '/v1.0/sites/{siteId}/structured-contents',
+	restSchema: 'StructuredContent',
+};
+
+const taxonomyVocabularyDataSetConfig = {
+	erc: getRandomString(),
+	label: getRandomString(),
+	restApplication: '/headless-admin-taxonomy/v1.0',
+	restEndpoint: '/v1.0/sites/{siteId}/taxonomy-vocabularies',
+	restSchema: 'TaxonomyVocabulary',
+};
 
 test.beforeEach(async ({dataSetManagerApiHelpers}) => {
 	dataSetERC = getRandomString();
 	dataSetLabel = getRandomString();
 
+	dataSetERCs.push(dataSetERC);
+
 	await dataSetManagerApiHelpers.createDataSet({
 		erc: dataSetERC,
 		label: dataSetLabel,
 	});
 });
 
-test.afterEach(async ({dataSetManagerApiHelpers}) => {
-	await dataSetManagerApiHelpers.deleteDataSet({erc: dataSetERC});
+test.afterEach(async ({apiHelpers, dataSetManagerApiHelpers}) => {
+	for (const DATA_SET_ERC of dataSetERCs) {
+		await dataSetManagerApiHelpers.deleteDataSet({
+			erc: DATA_SET_ERC,
+		});
+	}
+
+	dataSetERCs.length = 0;
+
+	if (article) {
+		await test.step('Move article to trash', async () => {
+			await apiHelpers.jsonWebServicesJournal.moveArticleToTrash(
+				siteId,
+				article.articleId
+			);
+		});
+	}
```

Comment:
Creation of web content seems like something that should be tested in `journal-web`. Do we need to include it in our tests?

Reply:
But the purpose of this test is to confirm that we can have a Data Set using the `StructuredContent` schema, so somehow we need to create sample data and do the clean up.

Reply:
We generally use our own DSM persistence endpoints in tests, as an easy/cheap way to assert DSM/FDS "works" with object-generated endpoints. I don't think it's a goal of these tests to assert full compatibility with all DXP endpoints, nevertheless, there is a point on using a wider variety of them in tests, under controlled conditions.

I'm still ok with using different ootb REST endpoints for testing from time to time. Though not the main focus for DSM/FDS testing in general, at the end of the day our tests are assuming some "default" characteristics are somehow homogeneusly present in the endpoints, namely:

- There is a `GET` call that returns a list of items 
- OpenAPI descriptionm for the endpoint is accurate enough
- Endpoint understands some query parameters like the ones controlling the item composition (e.g. `nestedFields`), pagination, filtering, sorting, etc
- Endpoint knows how to sort by any numeric, date and string fields
- Endpoint knows how to filter by any numeric, date and string fields

We do know that's not the case, there are inconsistencies/limitations. So we can't just pick any endpoint and expect a given DSM configuration will work. That's what makes these tests interesting.

I'd suggest to have a "healty" amount/presence of tests in those non-object, ootb endpoints. However if they end up being a  source of flakiness and maintenance work, I'm happy with reverting back to simple, object-based endpoints in testing.